### PR TITLE
Chore/typescript convert

### PR DESCRIPTION
**Pull request by james-gray-adaptive**

### CODE REVIEW

**Comment by UncleFirefox** [comment]
Hi! Do you think it might be worth it to add the tsc compilation command in travis.yml?

**Comment by bhavesh-desai-scratch** [comment]
I dont think you need to add the tsc compiler to travis/circle as compilation is done via webpack.  Webpack uses the the typescript library in the node_modules folder to compile.

File: src/client/src/services/model/spotPrice.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+              bid: Rate,
+              ask: Rate,
+              mid: Rate,
+              valueDate: Date,
+              creationTimestamp: Date,
+              priceMovementType: PriceMovementType,
+              spread: Spread,
+              isTradable: boolean) {
+    this._symbol = symbol;
+    this._ratePrecision = ratePrecision,
```
Comment (fdore) [inline src/client/src/services/model/spotPrice.ts]: typo: , => ;

Comment (james-gray-adaptive) [inline src/client/src/services/model/spotPrice.ts]: Thanks @fdore. I have resolved this.

**Pull request merged**